src/main/java/net/imglib2/Cursor.java

```diff
@@ -64,6 +64,12 @@
  * imglib Cursors, in general.
  * </p>
  *
+ * <p>
+ * It is not guaranteed that a Cursor will perform bounds checking. Therefore,
+ * moving a Cursor beyond the position of its last element will result in
+ * undefined behavior.
```

Comment:
I would modify to "...asking the position or value of a Cursor that was not advanced to its first element or was moved beyond its last element...". And possibly splitting that up into 2 sentences :-)